libraries/Microsoft.Bot.Builder.LanguageGeneration.Templates/ILanguageGenerator.cs

```diff
@@ -14,6 +14,6 @@ public interface ILanguageGenerator
         /// <param name="template">template or [templateId].</param>
         /// <param name="data">data to bind to.</param>
         /// <returns>text.</returns>
-        Task<string> Generate(ITurnContext turnContext, string template, object data);
+        Task<object> Generate(ITurnContext turnContext, string template, object data);
```

Comment:
>      Task<object> Generate(ITurnContext turnContext, string template, object data); [](start = 2, length = 84)

no, you should not be changing this.  The wholepoint of this interface is that it is binding to text.  There is a IActivityGenerator interface which is about generating activities.  In particular, I need to bind to language in places where I am not dealing with activities.  LG is broader then just activities.

Reply:
Currently the output of lg is an object, we need to extract the activity from this object. If we keep the string output of ILanguageGenerator, I'm ok. But another deserialization should be executed.